src/libraries/System.Console/tests/KeyParserTests.cs

```diff
@@ -402,6 +402,25 @@ public void BackTabEscapeSequence()
         Assert.Equal(ConsoleModifiers.Shift, consoleKeyInfo.Modifiers);
     }
 
+    [Theory]
+    [InlineData("test123", "test123")] // no bracketed paste
+    [InlineData("\u001B[200~test123\u001B[201~", "test123")] // simple bracketed paste with both start and end tag
+    [InlineData("\u001B[200~test123", "test123")] // no end tag
+    [InlineData("test123\u001B[201~", "test123")] // no start tag
+    [InlineData("\u001B[200~", "")] // just start tag
+    // the test case above are based on manual testing and observations
+    // the test cases below were not reproduced during manual testing, but hypothetically are possible
+    [InlineData("\u001B[201~", "")] // just end tag
+    [InlineData("\u001B[201~\u001B[200~test123", "test123")] // end tag (from previous pasted content) and start tag from next one
+    [InlineData("\u001B[201~\u001B[200~", "")] // same, but with no content
+    // Terminals that were tested don't allow for pasting empty clipboard to the Terminal.
+    // In case some allow for that, an empty content is returned and the caller must re-try read() call
+    [InlineData("\u001B[200~\u001B[201~", "")]
```

Comment:
Perhaps it would make sense to add one more case for nested start sequence:
```c#
    [InlineData("\e[200~nested \e[200~brackets \e[201~start \e[201~", "nested brackets start ")]
```
(trailing space is intentional)

Reply:
@am11 I tested nesting by pasting "nested" to the terminal, reading the input via `read()`, copying to clipboard with https://github.com/SimonCropp/TextCopy) and pasting to the Terminal again:

![image](https://user-images.githubusercontent.com/6011991/182557089-674cd275-53f8-4cce-a95e-2571d06c8b2e.png)

it seems that in such cases xterm is removing the escape characters from the "nested" tags. I can add a test case for that, but imo it would be a little bit odd (verifying that incomplete tags remain in the input).

```cs
[InlineData("\u001B[200~[200~nested[201~\u001B[201~", "[200~nested[201~ ")
```

but I can definitely add white spaces to show that they are not being trimmed.

Reply:
I added more test cases (whitespaces + nested)

Reply:
I was testing it on Ubuntu terminal, and noticed this:

```sh
$ printf '\u001B[200~nested \u001B[200~brackets \u001B[201~start \u001B[201~'
nested brackets start $
```

Reply:
@am11 I see. But this is something that you have printed to the Terminal, not something that you have read from it. Is it possible for the Terminal to return such data from `read`?

I am asking because with the current implementation we avoid copying memory and the code is relatively simple. I would prefer to keep it that way if there is no evidence that there is scenario that we don't support.

Reply:
Yes the terminal will remove the escape tag exactly the way this implementation does. If I am reading the change right, your implementation will remove the nested tag and extract the text in between just the way how terminal does it. That is why I was suggesting to test to preserve this behavior if it makes sense. Currently you added:

```
    [InlineData("\u001B[200~[200~nested[201~\u001B[201~", "[200~nested[201~")]
```

which is missing `\u001B` from the nested start tag so it is not testing the nested tag scenario. We may as well remove it since `[200~` without `\u001B` has no meaning.

> not something that you have read from it.

That was the intention. `printf` will send the raw string. To see what value is read, I used xxd to get the hex representation:

```sh
$ printf '\u001B[200~nested \u001B[200~brackets \u001B[201~start \u001B[201~' | xxd
00000000: 1b5b 3230 307e 6e65 7374 6564 201b 5b32  .[200~nested .[2
00000010: 3030 7e62 7261 636b 6574 7320 1b5b 3230  00~brackets .[20
00000020: 317e 7374 6172 7420 1b5b 3230 317e       1~start .[201~

# same with 'echo -e'
$ echo -e '\u001B[200~nested \u001B[200~brackets \u001B[201~start \u001B[201~' | xxd
00000000: 1b5b 3230 307e 6e65 7374 6564 201b 5b32  .[200~nested .[2
00000010: 3030 7e62 7261 636b 6574 7320 1b5b 3230  00~brackets .[20
00000020: 317e 7374 6172 7420 1b5b 3230 317e 0a    1~start .[201~.
```